drivers/media/platform/raspberrypi/rp1_cfe/cfe.c

```diff
@@ -250,7 +258,9 @@ struct cfe_node {
 	/* Pointer pointing to next v4l2_buffer */
 	struct cfe_buffer *next_frm;
 	/* Used to store current pixel format */
-	struct v4l2_format fmt;
+	struct v4l2_format vid_fmt;
+	/* Used to store current meta format */
+	struct v4l2_format meta_fmt;
```

Comment:
Why do we need both these since you can only have one of them populated?

Reply:
Good question. You _can_ have both of the populated. This is one of those v4l2 things again...

When you have a video node that can support both video and meta, you can call e.g. VIDIOC_S_FMT with type of V4L2_BUF_TYPE_VIDEO_CAPTURE, and VIDIOC_S_FMT with type of V4L2_BUF_TYPE_META_CAPTURE. Both will succeed.

It's only when you queue the buffers the type is "locked in". Until that, it's... both/neither =).

Having separate fields for vid and meta was the most sensible solution, which still makes v4l2-compliance happy, I was able to come up with.

Reply:
Thanks for the explanation.  Let's keep it as-is.